src/openrct2/peep/GuestPathfinding.h

```diff
@@ -37,19 +37,43 @@ extern RideId gPeepPathFindQueueRideIndex;
 // In practice, if this is false, gPeepPathFindQueueRideIndex is always RIDE_ID_NULL.
 extern bool gPeepPathFindIgnoreForeignQueues;
 
-// Given a peep 'peep' at tile 'loc', who is trying to get to 'gPeepPathFindGoalPosition', decide
-// the direction the peep should walk in from the current tile.
-Direction peep_pathfind_choose_direction(const TileCoordsXYZ& loc, Peep* peep);
+class GuestPathfinding
+{
+public:
+    virtual ~GuestPathfinding() = default;
 
-// Test whether the given tile can be walked onto, if the peep is currently at height currentZ and
-// moving in direction currentDirection.
-bool IsValidPathZAndDirection(TileElement* tileElement, int32_t currentZ, int32_t currentDirection);
+    // Given a peep 'peep' at tile 'loc', who is trying to get to 'gPeepPathFindGoalPosition', decide
+    // the direction the peep should walk in from the current tile.
+    virtual Direction peep_pathfind_choose_direction(const TileCoordsXYZ& loc, Peep* peep) = 0;
 
-// Overall guest pathfinding AI. Sets up Peep::DestinationX/DestinationY (which they move to in a
-// straight line, no pathfinding). Called whenever the guest has arrived at their previously set destination.
-//
-// Returns 0 if the guest has successfully had a new destination set up, nonzero otherwise.
-int32_t guest_path_finding(Guest* peep);
+    // Test whether the given tile can be walked onto, if the peep is currently at height currentZ and
+    // moving in direction currentDirection.
+    static bool IsValidPathZAndDirection(TileElement* tileElement, int32_t currentZ, int32_t currentDirection);
+
+    // Overall guest pathfinding AI. Sets up Peep::DestinationX/DestinationY (which they move to in a
+    // straight line, no pathfinding). Called whenever the guest has arrived at their previously set destination.
+    //
+    // Returns 0 if the guest has successfully had a new destination set up, nonzero otherwise.
+    virtual int32_t guest_path_finding(Guest* peep) = 0;
```

Comment:
```suggestion
    virtual int32_t CalculateNextDestination(Guest& peep) = 0;
```

Also, use javadoc format, e.g.:
```cpp
/**
 * Overall guest pathfinding AI. Sets up Peep::DestinationX/DestinationY (which they move to in a
 * straight line, no pathfinding). Called whenever the guest has arrived at their previously set destination.
 *
 * @returns 0 if the guest has successfully had a new destination set up, nonzero otherwise.
 */
```